### PR TITLE
Implement now for mingw/Windows

### PR DESCRIPTION
**Pull request by nprindle**
Implement `now` using `GetSystemTime` from `Win32`.

### CODE REVIEW

**Comment by chessai** [comment]
Awesome! This looks great.

You need to only depend on `Win32` if we're on windows. See as an example the cabal file of process: http://hackage.haskell.org/package/process-1.6.5.1/src/process.cabal

**Comment by chessai** [comment]
You'll also need to ifdef all imports therefrom, because Win32 can only compile on mingw

File: src/Chronos.hs [hunk 1]
```diff
@@ -248,7 +249,9 @@ import Data.Int (Int64)
 import Data.Char (isDigit)
 import Data.ByteString (ByteString)
 import Torsor (add,difference,scale,plus)
-import Chronos.Internal.CTimespec (getPosixNanoseconds)
```
Comment (chessai) [inline src/Chronos.hs]: Why is this import necessary?

Comment (nprindle) [reply src/Chronos.hs]: That was a typo from before switching to Win32.

**Comment by chessai** [comment]
The bounds on Win32 should be >= 2.2 && < 2.9. The API that we use has been available since the year 2000

**Comment by chessai** [comment]
Ok, this looks good to me right now

**Comment by chessai** [comment]
@ShrykeWindgrace how does this look?

**Comment by chessai** [comment]
Hmmm also does `now` return system UTC time, or system local time? I can't remember.

**Comment by nprindle** [comment]
`clock_gettime` with a `clock_id` of 0 should use `CLOCK_REALTIME` and return a UTC time, as far as I know. `Date.now()` from Javascript also returns a UTC time.

**Comment by chessai** [comment]
Yeah i just looked that up, sounds like everything's fine. I'll look at this more tomorrow

**Comment by chessai** [comment]
your branch still compiles for linux :+1:

**Comment by chessai** [comment]
i'm going to merge this, as it compiles under windows 10 for @nprindle, compiles for me on linux, and tests all still pass/haddocks still build. @ShrykeWindgrace could you please try this when you have the time? I'm sorry you hit this roadblock when trying to use chronos, it's just that none of us have windows machines and afaik no windows user has ever tried chronos out. Thanks!

**Comment by ShrykeWindgrace** [comment]
> 
> 
> @ShrykeWindgrace could you please try this when you have the time?
Just tried it, `stack test` is all green =) Sorry for the delay, yesterday was a public holiday and I did not have access to a windows machine either.

> I'm sorry you hit this roadblock when trying to use chronos, it's just that none of us have windows machines and afaik no windows user has ever tried chronos out. Thanks!

Not a problem at all, thanks for the fix! Windows users will definitely try it in the future (me, at least=) ), plus the `co-log` library recently migrated to `chronos`, too.
